packages/api-contract/src/base/Contract.ts

```diff
@@ -33,50 +34,52 @@ export type ContractCallResult<CallType extends ContractCallTypes> = CallType ex
 export default class Contract<ApiType extends ApiTypes> extends BaseWithTxAndRpcCall<ApiType> {
   public readonly address: Address;
 
-  constructor (api: ApiObject<ApiType>, abi: ContractABIPre | Abi, decorateMethod: DecorateMethod<ApiType>, address: string | AccountId | Address) {
+  constructor (api: ApiObject<ApiType>, abi: AnyJson | InkAbi, decorateMethod: DecorateMethod<ApiType>, address: string | AccountId | Address) {
     super(api, abi, decorateMethod);
 
-    this.address = this.registry.createType('Address', address);
+    this.address = this.api.registry.createType('Address', address);
```

Comment:
It should point to the own registry here, it always needs to in any type. The API one is not explicitly required to point to latest. It does move around based on the actual queries we are making, i.e. non-static and known types may be removed.